### DISCUSSION TITLE
`goreleaser release --nightly` doesn't publish anything

### DISCUSSION THREAD

**Discussion by rfay** [goreleaser/goreleaser#3194] [Ideas, Issue Triage, and General discussions]
I'm sure there's something fundamental I don't understand, but trying out `goreleaser release --nightly --rm-dist` gets me nothing published. 

At the end I see

```
   • calculating checksums
   • storing release metadata
      • writing                   file=dist/artifacts.json
      • writing                   file=dist/metadata.json
   • publishing
   • release succeeded after 2.70s
   • thanks for using goreleaser-pro!
```

but note nothing happens in the publishing stage.

* Full [output](https://gist.github.com/rfay/3404b00d894d9a555cbafafde074eb5a)
* [.goreleaser.yml](https://github.com/rfay/ddev/blob/20220623_goreleaser_release/.goreleaser.yml)

**Comment by caarlos0** [comment]
Yes, its expected, it only builds everything, but the whole publishing phase is skipped:


https://goreleaser.com/customization/nightlies/ 


You can then docker push only what you want and upload artifacts using, for instance, `actions/upload-artifact`.

This probably need more docs and better examples though, added to my todo list.

**Reply by rfay** [reply]
Oh dear, I thought that it would maintain a nightly release on github. That's kind of what people would expect I think. actions/upload-artifact isn't up to the task because it can only create new releasees which would make a mess. Why is this a feature of pro then, when anybody could just do `goreleaser release --snapshot` ?

**Reply by rfay** [reply]
What I'd like to do is maintain a HEAD or 'latest' release in github releases, which always has latest built the same as a release builds them.

**Reply by caarlos0** [reply]
That's something I'll work on at some time, but not available today.

That said, snapshots and nightlies have a few differences in what they build, version numbers, what is skipped or not, etc.

**Comment by caarlos0** [comment]
This has been changed over the years, it should now publish a nightly release

closing
